crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -0,0 +1,389 @@
+                markup! {
+                    "A \"floating\" Promise was found, meaning it is not properly handled and could lead to ignored errors or unexpected behavior."
+                },
+            )
+            .note(markup! {
+                "This happens when a Promise is not awaited, lacks a `.catch` or `.then` rejection handler, or is not explicitly ignored using the `void` operator."
+            })
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let expression = node.expression().ok()?;
+        let mut mutation = ctx.root().begin();
+
+        let await_expression = AnyJsExpression::JsAwaitExpression(make::js_await_expression(
+            make::token(JsSyntaxKind::AWAIT_KW)
+                .with_trailing_trivia([(TriviaPieceKind::Whitespace, " ")]),
+            expression.clone().trim_leading_trivia()?,
+        ));
+
+        mutation.replace_node(expression, await_expression);
+        Some(JsRuleAction::new(
+            ctx.metadata().action_category(ctx.category(), ctx.group()),
+            ctx.metadata().applicability(),
+            markup! { "Add await operator." }.to_owned(),
+            mutation,
+        ))
+    }
+}
+
+/// Checks if the callee of a JavaScript expression is a promise.
+///
+/// This function inspects the callee of a given JavaScript expression to determine
+/// if it is a promise. It returns `true` if the callee is a promise, otherwise `false`.
+///
+/// The function works by finding the binding of the callee and checking if it is a promise.
+///
+/// # Arguments
+///
+/// * `callee` - A reference to an `AnyJsExpression` representing the callee to check.
+/// * `model` - A reference to the `SemanticModel` used for resolving bindings.
+///
+/// # Returns
+///
+/// * `true` if the callee is a promise.
+/// * `false` otherwise.
+///
+/// # Examples
+///
+/// Example JavaScript code that would return `true`:
+/// ```typescript
+/// async function returnsPromise(): Promise<string> {
+///     return "value";
+/// }
+///
+/// returnsPromise().then(() => {});
+/// ```
+///
+/// Example JavaScript code that would return `false`:
+/// ```typescript
+/// function doesNotReturnPromise() {
+///     return 42;
+/// }
+///
+/// doesNotReturnPromise().then(() => {});
+/// ```
+fn is_callee_a_promise(callee: &AnyJsExpression, model: &SemanticModel) -> bool {
+    match callee {
+        AnyJsExpression::JsIdentifierExpression(ident_expr) => {
+            let Some(reference) = ident_expr.name().ok() else {
+                return false;
+            };
+
+            let Some(binding) = model.binding(&reference) else {
+                return false;
+            };
+
+            let Some(any_js_binding_decl) = binding.tree().declaration() else {
+                return false;
+            };
+
+            let AnyJsBindingDeclaration::JsFunctionDeclaration(func_decl) = any_js_binding_decl
```

Comment:
- For now, only checking a function declaration
- Not considering reassigned variables like this
```
async function returnsPromise(): Promise<string> {
  return 'value';
}

let reassigned = returnsPromise;
reassigned();
```

Reply:
Thank you for letting me know! 
I’ll revisit this when I work on implementing support for other functions or methods